x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java

```diff
@@ -1891,6 +1891,17 @@ public void testLookup() {
         assertThat(e.getMessage(), containsString("LOOKUP not yet supported"));
     }
 
+    public void testImplicitCasting() {
+        var e = expectThrows(VerificationException.class, () -> analyze("""
```

Comment:
Add another test that adds two strings together that look like a date/time:
`2024/03/02" + "1 day"